Fix overflow labels in recursive-simple tasks

MartinSpiessl opened the pull request:
Hi,

I found 3 benchmark files with wrong overflow label:

 c/recursive-simple/id_b3_o2_false-unreach-call_true-termination_true-no-overflow.c
 c/recursive-simple/id_b3_o5_true-unreach-call_true-termination_true-no-overflow.c
 c/recursive-simple/id_b5_o10_true-unreach-call_true-termination_true-no-overflow.c

These were added recently in commit a45f1894470dc5433310cea27d185fdc8f8bda48 with the default label stating that there are no overflows. On inspection it is obvious that all of them contain (the same) overflow. For example see id_b3_o2_false-unreach-call_true-termination_true-no-overflow.c with comments pointing out the overflows:

```
extern int __VERIFIER_nondet_int();
extern void __VERIFIER_error();

int id(int x) {
  if (x==0) return 0;
  int ret = id(x-1) + 1; // overflow if x=INT_MIN
  if (ret > 3) return 3;
  return ret;
}

int main(void) {
  int input = __VERIFIER_nondet_int(); // input = INT_MIN leads to overflow
  int result = id(input);
  if (result == 2) {
    ERROR: __VERIFIER_error();
  }
}
```

This pull request will:
- fix the overflows in the files so that the current label is correct 
- add a copy of the unmodified files with the label updated to false-no-overflow

kfriedberger commented:
Overflow is undefined behaviour and should not be combined with any other label.
I would suggest to not rename the tasks, but change the internal condition "x==0" into "x<=0", which should avoid the overflow.

PhilippWendler commented:
If a task is fixed, the original task should also be kept and labeled appropriately.

kfriedberger commented:
Ok, tasks should be valid now.

MartinSpiessl commented:
@kfriedberger oh yes I forgot to remove the other labels, I just fixed that. I did not rename the tasks. Instead i fixed the file by adding a cast to unsigned int. This will fix the file without changing the behavior, because casting from signed int to unsigned int and then back to signed int is well defined in the C standard.

@PhilippWendler  original tasks are kept and labels should now be appropriate
